firestore-bundle-server/functions/__tests__/functions.test.ts

```diff
@@ -0,0 +1,286 @@
+
+    /*** check document */
+    expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  it("successfully returns a bundle using a query with a collection", async () => {
+    const bundleName = "query-with-a-collection";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    /*** check metadata */
+    expect(metadata.metadata.id).toEqual(bundleName);
+    expect(metadata.metadata.totalDocuments).toEqual(2);
+
+    /*** check document metadata */
+    expect(documentMetadata.namedQuery.name).toEqual("example");
+
+    /*** check document */
+    expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  it("successfully returns a bundle using a query with a collection and condition", async () => {
+    const bundleName = "query-with-a-collection-and-condition";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    /*** check metadata */
+    expect(metadata.metadata.id).toEqual(bundleName);
+    expect(metadata.metadata.totalDocuments).toEqual(1);
+
+    /*** check document metadata */
+    expect(documentMetadata.namedQuery.name).toEqual("example");
+
+    /*** check document */
+    expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  it("successfully returns a bundle using a query with a collection and where clause", async () => {
+    const bundleName = "query-with-a-collection-and-condition";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    /*** check metadata */
+    expect(metadata.metadata.id).toEqual(bundleName);
+    expect(metadata.metadata.totalDocuments).toEqual(1);
+
+    /*** check document metadata */
+    expect(documentMetadata.namedQuery.name).toEqual("example");
+
+    /*** check document */
+    expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  xit("successfully returns a bundle using a query with a collection and multiple where clauses", async () => {
+    const bundleName = "query-with-a-collection-and-multiple-where-conditions";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    /*** check metadata */
+    expect(metadata.metadata.id).toEqual(bundleName);
+    expect(metadata.metadata.totalDocuments).toEqual(1);
+
+    /*** check document metadata */
+    expect(documentMetadata.namedQuery.name).toEqual("example");
+
+    /*** check document */
+    expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  xit("successfully returns a bundle using a document", async () => {
```

Comment:
`xit` on a few of these?

Reply:
I've added these as placeholders for now. Some errors timeouts appear on a couple the tests, so I wanted to highlight  these before continuing developing them.